Example Apps/Example Swift/MenuManager.swift

```diff
@@ -10,12 +10,29 @@ import Foundation
 import SmartDeviceLink
 import SmartDeviceLinkSwift
 
+typealias RefreshTemplateHandler = ((SDLPredefinedLayout) -> Void)
+
 class MenuManager: NSObject {
+    fileprivate var refreshTemplateHandler: RefreshTemplateHandler?
+
+    /// Setting current selected template
+    public fileprivate(set) var currentTemplate: SDLPredefinedLayout {
+        didSet {
+            guard let refreshTemplateHandler = refreshTemplateHandler else { return }
+            refreshTemplateHandler(currentTemplate)
+        }
+    }
+
+    init(updateTemplate: RefreshTemplateHandler? = nil) {
+        self.refreshTemplateHandler = updateTemplate
+        currentTemplate = .nonMedia
+    }
+
     /// Creates and returns the menu items
     ///
     /// - Parameter manager: The SDL Manager
     /// - Returns: An array of SDLAddCommand objects
-    class func allMenuItems(with manager: SDLManager, choiceSetManager: PerformInteractionManager) -> [SDLMenuCell] {
+    func allMenuItems(with manager: SDLManager, choiceSetManager: PerformInteractionManager) -> [SDLMenuCell] {
```

Comment:
```suggestion
    var allVoiceMenuItems: [SDLVoiceCommand] {
```

Reply:
Since we can remove the parameters after passing the `sdlManager` and the `choiceSetManager` in the `init`, I think it makes more sense for each of the menu item functions to be refactored into variables.